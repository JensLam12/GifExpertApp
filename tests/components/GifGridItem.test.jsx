import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components";

describe( 'Pruebas en <GifItem />', () => {

    const title = 'Saitama';
    const url = 'https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExOWIwMThlNmQ2MjhlNmFlZmE2NmFhNDIzNDc0MThmYjNmNDk5MDU4ZCZjdD1n/yo3TC0yeHd53G/giphy.gif';

    test( 'should be match with snapshot', () => {
        const { container } = render(
            <GifGridItem
                title = {title}
                url={url}
            />
        );
        expect(container).toMatchSnapshot();
    });

    test( 'should show image with url and indicated alt', () => {
        render( <GifGridItem title = {title} url={url} /> );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test( 'should show title in component', () => {
        render( <GifGridItem title = {title} url={url} /> );
        expect(screen.getByText(title)).toBeTruthy();
    });

});