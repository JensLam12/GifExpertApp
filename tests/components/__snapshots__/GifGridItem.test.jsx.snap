// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> should be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExOWIwMThlNmQ2MjhlNmFlZmE2NmFhNDIzNDc0MThmYjNmNDk5MDU4ZCZjdD1n/yo3TC0yeHd53G/giphy.gif"
      src="https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExOWIwMThlNmQ2MjhlNmFlZmE2NmFhNDIzNDc0MThmYjNmNDk5MDU4ZCZjdD1n/yo3TC0yeHd53G/giphy.gif"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
